Java/libraries/recognizers-text-number/src/main/java/com/microsoft/recognizers/text/number/models/AbstractNumberModel.java

```diff
@@ -54,7 +54,7 @@ protected AbstractNumberModel(IParser parser, IExtractor extractor) {
             return new ModelResult(
                     o.getText(),
                     o.getStart(),
-                o.getStart() + o.getLength(),
+                o.getStart() + o.getLength() - 1,
```

Comment:
Can you add a code comment before to return to specify this is to keep behaviour consistent with the other platforms?
Also, how did the *Model.json unit tests pass if this was incorrect?
Can you fix whatever issue there is there in this PR too? Thanks!

Reply:
Hi @tellarin, we already added the code comment before to `return` as requested.

Also, we noticed that the `start` and `end` were not validated and that's why the tests were passing with/without these changes. We updated the `NumberTest` following C# and JavaScript in order to validate the necessary properties divided in each type of Recognizer.

We confirmed that all the tests are correctly passing 😊.

Reply:
Thanks! I'll review soon. Are the other models correct? DateTime, Units, ...?

Reply:
@tellarin, all the models are validating the `typeName`, `text` and `Resolution` properties of the specs.

If you are okay, we can create an issue improving this behavior for all the models apart from this PR/issue in order to have this PR reviewed/merged as soon as possible 😊.

Reply:
Please create the issue and a new PR. 😊